### PR TITLE
verify-owners: extract parsing, improve tests

### LINKED ISSUE #8838: Improve test coverage of verify-owners
Suggested in #8806 
- add tests with patch to check review line number
- factor out the parse/generate `messageWithLine` logic and unit-test it

### PR DESCRIPTION
**Pull request by matthyx**
fixes #8838

### CODE REVIEW

File: prow/plugins/verify-owners/verify-owners_test.go [hunk 1]
```diff
@@ -46,22 +48,22 @@ labels:
     labels:
     - label1
 `),
-	"invalidSyntax": []byte(`approvers
+	"invalidSyntax": []byte(`approvers:
```
Comment (stevekuznetsov) [inline prow/plugins/verify-owners/verify-owners_test.go]: Why did these have to change?

File: prow/plugins/verify-owners/verify-owners_test.go [hunk 2]
```diff
@@ -292,3 +294,134 @@ func TestHandle(t *testing.T) {
...
+			line: 7,
+		},
+		{
+			name: "invalidSyntaxFilters",
+			err:  true,
+			line: 9,
+		},
+		{
+			name:  "invalidSyntaxFilters edit",
+			patch: "@@ -5,6 +5,6 @@ filters:\n     reviewers:\n     - alice\n     - bob\n-    labels\n+    label\n     - label1\n",
```
Comment (stevekuznetsov) [inline prow/plugins/verify-owners/verify-owners_test.go]: Can we use grave accents (back-ticks) to format this nicely as a multiline string?

File: prow/plugins/verify-owners/verify-owners_test.go [hunk 3]
```diff
@@ -292,3 +294,134 @@ func TestHandle(t *testing.T) {
 		}
 	}
 }
+
+func TestParseOwnersFile(t *testing.T) {
```
Comment (stevekuznetsov) [inline prow/plugins/verify-owners/verify-owners_test.go]: Shouldn't we also make this test the actual parsed output?

File: prow/plugins/verify-owners/verify-owners_test.go [hunk 4]
```diff
@@ -292,3 +294,134 @@ func TestHandle(t *testing.T) {
...
+	cases := []struct {
+		name  string
+		patch string
+		err   bool
+		line  int
+	}{
+		{
+			name: "emptyApprovers",
```
Comment (fejta) [inline prow/plugins/verify-owners/verify-owners_test.go]: I assume this is causing some magic to load an emptyApprovers document? I would prefer we inline the document, so I don't have to understand how we fetch it.

File: prow/plugins/verify-owners/verify-owners_test.go [hunk 5]
```diff
@@ -292,3 +294,134 @@ func TestHandle(t *testing.T) {
 		}
 	}
 }
+
+func TestParseOwnersFile(t *testing.T) {
+	cases := []struct {
+		name  string
+		patch string
+		err   bool
+		line  int
```
Comment (fejta) [inline prow/plugins/verify-owners/verify-owners_test.go]: can we call this errLine?

File: prow/plugins/verify-owners/verify-owners_test.go [hunk 6]
```diff
@@ -292,3 +294,134 @@ func TestHandle(t *testing.T) {
 		}
 	}
 }
+
+func TestParseOwnersFile(t *testing.T) {
+	cases := []struct {
+		name  string
+		patch string
+		err   bool
```
Comment (fejta) [inline prow/plugins/verify-owners/verify-owners_test.go]: nit: given that nearly all scenarios have an error, I'd switch this to pass bool

File: prow/plugins/verify-owners/verify-owners_test.go (hunk 3 above)
Comment (fejta) [reply prow/plugins/verify-owners/verify-owners_test.go]: You mean the message? If so please encapsulate all the message in either consts or functions. Not entirely convinced its necessary

File: prow/plugins/verify-owners/verify-owners_test.go (hunk 4 above)
Comment (stevekuznetsov) [reply prow/plugins/verify-owners/verify-owners_test.go]: It's using the zero value in the struct. but I agree

File: prow/plugins/verify-owners/verify-owners_test.go (hunk 1 above)
Comment (matthyx) [reply prow/plugins/verify-owners/verify-owners_test.go]: It was to make the invalid line more to the end of the file in order to have a patch not starting with line 1.

File: prow/plugins/verify-owners/verify-owners_test.go (hunk 6 above)
Comment (matthyx) [reply prow/plugins/verify-owners/verify-owners_test.go]: actually I can use errLine = 0 instead

File: prow/plugins/verify-owners/verify-owners_test.go (hunk 3 above)
Comment (matthyx) [reply prow/plugins/verify-owners/verify-owners_test.go]: @stevekuznetsov your call...

Comment (stevekuznetsov) [reply prow/plugins/verify-owners/verify-owners_test.go]: No no, what Erick said is right

**Pull request merged**